evaluation/metrics/train_file_metrics.py

```diff
@@ -0,0 +1,49 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from sklearn.multioutput import MultiOutputClassifier
+from sklearn.svm import LinearSVC
+
+
+def dataperf(train, test, constructor_args):
```

Comment:
@TristanThrush can this function be modified to handle N train set for each label that we are evaluating on instead of one multi-label dataset? Also can we also handle N test sets for each label as well?

Something like 
`train: Dict[str, Dict[str, int]] = {"DOG": {"id_0": 1,
                                                                      "id_1": 0,
                                                                      "id_3": 1},
                                                       "CAT": {"id_0": 1,
                                                                      "id_2": 1,
                                                                      "id_3": 0}}`
and the same for the test dataset

Reply:
Yes, I can definitely train up n binary classifiers in here. Given the architecture of our system, though, I think that it could make the most sense to keep the input formats to the dataperf function the same as they are. This function is already called as a subroutine for each dataset you add, and you can add however many datasets you want. Is that ok?

Reply:
If we train N-binary classifiers we would need to make a distinction at least in the train parameter to understand which sample is part of a labeled train set. The reason being we would need to distinguish between positive and negative samples for each labeled training set.

Reply:
You can take this (the existing input format):
```
{{"uid": "train_id_0", "labels": ["DOG", "CAT"]}, {"uid": "train_id_1", "labels": ["CAT"]}, {"uid": "train_id_2", "labels": ["DOG"]}}
```
and convert it into this (your desired format):
```
{"DOG": {"train_id_0": 1, "train_id_1": 0, "train_id_3": 1}, "CAT": {"train_id_0": 1, "train_id_2": 1, "train_id_3": 0}}
```
I can do this conversion within the dataperf function itself. I am only suggesting that we leave the input format to the dataperf function the same. Is that ok?

Reply:
Btw, I believe this is already what I do, with multioutput classifier. https://scikit-learn.org/stable/modules/generated/sklearn.multioutput.MultiOutputClassifier.html#sklearn.multioutput.MultiOutputClassifier

Reply:
Mutlioutput classifier is n binary classifiers, according to the documentation

Reply:
If every example has a 0 or 1 for every label, then the absence of "DOG" in the labels list means that it is a negative example for dog.

Reply:
Each train set for a label can have different ids. All of the ids that are in a train set for a label will have a positive or negative sample. The reason being is that for each label we want to find a small dataset (say 100 samples) that is representative to perform the classification task at hand. 

Since we are testing different selection strategies here and evaluating the performance. Say we have a dataset of 1M samples and our selection strategy selects 100 high value samples from the 1M sample dataset. It could be that when selecting sample for a DOG train set the selection strategy finds different samples than a CAT train set. 

The point here is to test an active learning/data selection strategy that can find samples that leads to good performance for the task at hand (binary classification). This has good implications on labeling costs and iteration time for models if we can determine if a selection strategy that will require orders of magnitudes less samples to label can perform as well if not better than training on a full labeled dataset.

Reply:
Ok so I gather that labels can be undefined, yes? I think that we found the source of the confusion, and I'll work on a change where we only train the binary classifier for "cat" on examples where we explicitly tell it "no cat" or "cat".

Reply:
Thanks for clarifying this

Reply:
No problem. Let me know if you want to sync about how to set this up.